reactful-numbers/src/app/components/CalculatorBox.js

```diff
@@ -1,12 +1,37 @@
+"use client";
 import React from "react";
 import ExpressionsBox from "./ExpressionsBox";
 import ButtonsBox from "./ButtonsBox";
-
+import { useState } from "react";
 function CalculatorBox() {
+  const [inputValue, setInputValue] = useState("");
+  const handleButtonClick = (text) => {
+    setInputValue(inputValue + text);
+  };
+  const clearInput = () => {
+    setInputValue("");
+  };
+  const invertSign = () => {
+    setInputValue(inputValue * -1);
+  };
+  const percentage = () => {
+    setInputValue(inputValue / 100);
+  };
+  const equals = () => {
+    setInputValue(eval(inputValue));
+  };
+
   return (
     <div className="calculator-box">
-      <ExpressionsBox></ExpressionsBox>
-      <ButtonsBox></ButtonsBox>
+      <ExpressionsBox value={inputValue}></ExpressionsBox>
+      <ButtonsBox
```

Comment:
Allowing inverse data flow from components lower in the hieararchy to components that are up by providing the components down with functions that change the state of the data.